### PR TITLE
Support Aws kinesis V1 and V2 SDKs

### PR DESCRIPTION
**Pull request by obenkenobi**
### Overview
Add support for adding external cloud traces for customers using AWS Kinesis v1 and v2 SDKs

### Related Github Issue
https://github.com/newrelic/newrelic-java-agent/issues/2016

### CODE REVIEW

File: instrumentation/aws-java-sdk-kinesis-1.11.106/src/main/java/com/agent/instrumentation/awsjavasdk2/services/kinesis/KinesisUtil.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        }
+    }
+
+    public static void setTraceDetails(String kinesisOperation, TracedMethod tracedMethod) {
+        tracedMethod.setMetricName(TRACE_CATEGORY, kinesisOperation);
+        tracedMethod.reportAsExternal(createCloudParams());
+    }
+
+    public static CloudParameters createCloudParams() {
+        // Todo: add arn to cloud parameters
```
Comment (jbedell-newrelic) [inline instrumentation/aws-java-sdk-kinesis-1.11.106/src/main/java/com/agent/instrumentation/awsjavasdk2/services/kinesis/KinesisUtil.java:59]: Is this still a TODO?  Is this method necessary at just 1 line?

File: instrumentation/aws-java-sdk-kinesis-2.0.6/src/main/java/com/agent/instrumentation/awsjavasdk2/services/kinesis/KinesisUtil.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+        return segment;
+    }
+
+    public static void setTraceDetails(String kinesisOperation) {
+        TracedMethod tracedMethod = NewRelic.getAgent().getTracedMethod();
+        tracedMethod.setMetricName(TRACE_CATEGORY, kinesisOperation);
+        tracedMethod.reportAsExternal(createCloudParams());
+    }
+    public static CloudParameters createCloudParams() {
+        // Todo: add arn to cloud parameters
```
Comment (jbedell-newrelic) [inline instrumentation/aws-java-sdk-kinesis-2.0.6/src/main/java/com/agent/instrumentation/awsjavasdk2/services/kinesis/KinesisUtil.java:26]: Is this still a TODO?  Is this method necessary at just 1 line?

File: instrumentation/aws-java-sdk-kinesis-1.11.106/src/main/java/com/agent/instrumentation/awsjavasdk2/services/kinesis/KinesisUtil.java (hunk 1 above)
Comment (obenkenobi) [reply instrumentation/aws-java-sdk-kinesis-1.11.106/src/main/java/com/agent/instrumentation/awsjavasdk2/services/kinesis/KinesisUtil.java:59]: Yes, there is a separate [ticket](https://github.com/newrelic/newrelic-java-agent/issues/2017) for reporting the ARN for kinesis, I am simply adding the todo for when I begin this ticket.

File: instrumentation/aws-java-sdk-kinesis-2.0.6/src/main/java/com/agent/instrumentation/awsjavasdk2/services/kinesis/KinesisUtil.java (hunk 2 above)
Comment (obenkenobi) [reply instrumentation/aws-java-sdk-kinesis-2.0.6/src/main/java/com/agent/instrumentation/awsjavasdk2/services/kinesis/KinesisUtil.java:26]: This is a repeat of what I said in another comment:
Yes, there is a separate [ticket](https://github.com/newrelic/newrelic-java-agent/issues/2017) for reporting the ARN for kinesis, I am simply adding the todo for when I begin this ticket.

**Review by jbedell-newrelic — APPROVED** [review]

**Pull request merged**